python/src/QuantileConfidence_doc.i.in

```diff
@@ -0,0 +1,423 @@
+This class estimates bounds of the quantile of level :math:`\alpha \in [0,1]` of the random variable :math:`X`
+with a confidence greater than :math:`\beta`, using a given order statistics.
+
+Let :math:`x_{\alpha}` be the unknown quantile of level :math:`\alpha` of the random variable :math:`X` of dimension 1.
+Let :math:`(X_1, \dots, X_\sampleSize)` be a sample of independent and identically distributed variables according to :math:`X`.
+The bounds of the interval are computed from order statistics that we now introduce.
+Let :math:`X_{(k)}` be the :math:`k` -th order statistics of :math:`(X_1, \dots, X_\sampleSize)` which means that
+:math:`X_{(k)}` is the :math:`k` -th maximum of :math:`(X_1, \dots, X_\sampleSize)` for :math:`1 \leq k \leq \sampleSize`.
+For example, :math:`X_{(1)} = \min (X_1, \dots, X_\sampleSize)` is the minimum
+and :math:`X_{(\sampleSize)} = \max (X_1, \dots, X_\sampleSize)` is the maximum. We have:
+
+.. math::
+
+    X_{(1)} \leq X_{(2)} \leq \dots \leq X_{(\sampleSize)}
+
+
+Examples
+--------
+>>> import openturns as ot
+>>> import openturns.experimental as otexp
+>>> alpha = 0.05
+>>> beta = 0.95
+>>> sample = ot.Gumbel().getSample(100)
+>>> algo = otexp.QuantileConfidence(alpha, beta)
+>>> ci = algo.computeUnilateralConfidenceInterval(sample)
+"
+
+// ---------------------------------------------------------------------
+
+%feature("docstring") OT::QuantileConfidence::computeUnilateralRank
+"Evaluate an unilateral rank of a quantile.
+
+The lower tail rank is the largest integer :math:`k_{low}` such as
+
+.. math::
+
+    \Pset(X_{(k)} \leq x_{\alpha}) \geq \beta
+
+and is given by:
+
+.. math::
+
+    k_{low} = \overline{F}^{-1}_{n, \alpha}(\beta)
+
+which exists only if
+
+.. math::
+
+    1-(1-\alpha)^n \ge \beta
+
+The upper tail rank is the smallest integer :math:`k_{up}` such as
+
+.. math::
+
+    \Pset(x_{\alpha} \leq X_{(k)}) \geq \beta
+
+and is given by:
+
+.. math::
+
+    k_{up} = F^{-1}_{n, \alpha}(\beta)
+
+which exists only if
+
+.. math::
+
+    1-\alpha^n \geq \beta, \beta > 0
+
+Parameters
+----------
+size : int
+    Sample size
+tail : bool, optional
+    Whether to return the lower or upper tail rank
+
+Returns
+-------
+rank : int
+    Rank :math:`k \in \{0, \hdots, n-1\}`
+
+Examples
+--------
+>>> import openturns as ot
+>>> import openturns.experimental as otexp
+>>> alpha = 0.05
+>>> beta = 0.95
+>>> algo = otexp.QuantileConfidence(alpha, beta)
+>>> rank = algo.computeUnilateralRank(100)
+"
+
+// ---------------------------------------------------------------------
+
+%feature("docstring") OT::QuantileConfidence::computeBilateralRank
+"Evaluate the bilateral rank of a quantile.
+
+The bilateral ranks :math:`k1, k2` are the integers minimizing the probability
+
+.. math::
+
+    \argmin_{k1, k2 \in \{0, \hdots, n-1\}^2\, | \, \Pset(k_1 < X \leq k_2) \geq \beta} \Pset(k_1 < X \leq k_2) | X \sim \cB(n, \alpha)
+
+which exists only if
+
+.. math::
+
+    1 - \alpha^n - (1-\alpha)^n \ge \beta
```

Comment:
Here is the rendering:

![image](https://github.com/user-attachments/assets/38cc5175-42a2-4de7-8056-995502df17d0)

```suggestion
The bilateral ranks :math:`k1, k2` are the integers minimizing the probability

.. math::

    \argmin_{0 \leq k_1 \leq k2 \leq n - 1 \, | \, \Pset(k_1 < X \leq k_2) \geq \beta} \Pset(k_1 < X \leq k_2) | 

where :math:`X \sim \cB(n, \alpha)`.
These ranks exist only if:

.. math::

    1 - \alpha^n - (1-\alpha)^n \ge \beta.
```